src/classes/queue-scheduler.ts

```
@@ -53,97 +53,94 @@ export class QueueScheduler extends QueueBase {
-            this.nextTimestamp = nextTimestamp;
+          if (data && data[0]) {
+            const stream = data[0];
+            const events = stream[1];
+
+            for (let i = 0; i < events.length; i++) {
+              streamLastId = events[i][0];
+              const args = array2obj(events[i][1]);
+              const nextTimestamp: number = parseInt(args.nextTimestamp);
+
+              if (nextTimestamp < this.nextTimestamp) {
+                this.nextTimestamp = nextTimestamp;
+              }
+            }
+
+            //
+            // We trim to a length of 100, which should be a very safe value
+            // for all kind of scenarios.
+            //
+            if (!this.closing) {
+              await client.xtrim(key, 'MAXLEN', '~', 100);
+            }
           }
-        }
-
-        //
-        // We trim to a length of 100, which should be a very safe value
-        // for all kind of scenarios.
-        //
-        if (!this.closing) {
-          await client.xtrim(key, 'MAXLEN', '~', 100);
-        }
-      }
 
-      const now = Date.now();
-      const delay = this.nextTimestamp - now;
-
-      if (delay <= 0) {
-        const [nextTimestamp, id] = await this.updateDelaySet(now);
-        if (nextTimestamp) {
-          this.nextTimestamp = nextTimestamp;
-          streamLastId = id;
-        } else {
-          this.nextTimestamp = Number.MAX_VALUE;
+          const now = Date.now();
+          const delay = this.nextTimestamp - now;
+
+          if (delay <= 0) {
+            const [nextTimestamp, id] = await this.updateDelaySet(now);
+            if (nextTimestamp) {
+              this.nextTimestamp = nextTimestamp;
+              streamLastId = id;
+            } else {
+              this.nextTimestamp = Number.MAX_VALUE;
+            }
+          }
         }
+        this.running = false;
+      } catch (error) {
+        this.running = false;
+        console.error(error);
```

Instead of console.error here we should just throw the exception so the user calling this method should handle it